other/utils.py

```diff
@@ -392,7 +395,6 @@ def server_main(name, server_class, server_handler):
             thread.join(0.1)  # Timeout allows main thread to handle signals
             if args.log_to_window:
                 ui_update()
-    except KeyboardInterrupt:
+    except (KeyboardInterrupt, Exception):
         server.info("Interrupt key was pressed, closing server...")
-        server.shutdown()
-        server.server_close()
+        server.close()
```

Comment:
What about adding another `except Exception` block and logging the exception via `server.error` then with a `finally` statement call `server.close`?
```python
try:
    # (...)
except KeyboardInterrupt:
    server.info("Interrupt key was pressed, closing server...")
except Exception:
    server.error(...)
finally:
    server.close()
```